### ISSUE TITLE
Clarify style for C++ functions that return pointers

### ISSUE THREAD

**Issue by jamesbraza** [google/styleguide#609]
One area of the [C++ style guide](https://google.github.io/styleguide/cppguide.html) that could use clarification is functions that return pointers.

I have seen all three of these variants before:

```cpp
int* Foo() { }
int * Bar() { }
int *Baz() { }
```

I think it would be simple to update the style guide in either the [Return Values](https://google.github.io/styleguide/cppguide.html#Return_Values) or the [Pointer or Reference Expressions](https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions) sections to include the suggested method.

**Comment by tituswinters** [comment]
I think it's covered by the Pointer or Reference Expressions section: you can have the * or & on the left or the right, but be consistent. Spaces on both side are forbidden.

Reopen if you don't think that's clear enough.

**Comment by jamesbraza** [comment]
Yeah I had seen that, and thought it was _halfway_ appropriate, if you're talking about this part:

```cpp
// These are fine, space preceding.
char *c;
const std::string &str;

// These are fine, space following.
char* c;
const std::string& str;
```

From my understanding, that's for pointer and reference variable _initialization_, not the when a function returns a pointer.

---

I created this issue because I keep seeing all variants of pointer return.  Here are some examples:

- `*` on the left side: https://stackoverflow.com/a/41869925/11163122
- Spaces on both sides of `*`: https://stackoverflow.com/a/19042634/11163122
- `*` on the right side: https://stackoverflow.com/a/7754999/11163122

I figured maybe it would be good for Google's Style Guide to address this.

---

What could be interpreted to be the proper formatting is implied in Bjarne Stroustrup's [C++ Style and Technique FAQ in the memory leaks subsection](https://www.stroustrup.com/bs_faq2.html#memory-leaks):

```cpp
S* f()
{
  return new S;	// who is responsible for deleting this S?
};
```

**Comment by tituswinters** [comment]
That's the first I've seen of there being any confusion, but you're right that the text doesn't strictly say this. I'll see if I can clarify without it getting too wordy or messy.

**Comment by tituswinters** [comment]
This'll be resolved the next time we push an update. The revised text now reads:
"When referring to a pointer or reference (variable declarations or definitions, arguments,
return types, template parameters, etc), you may place the space before or after the
asterisk/ampersand. In the trailing-space style, the space is elided in some cases (template
parameters, etc)."
(With updated examples.)
